gattc_linux.go

```diff
@@ -91,6 +108,8 @@ func (d *Device) DiscoverServices(uuids []UUID) ([]DeviceService, error) {
 // DeviceCharacteristic is a BLE characteristic on a connected peripheral
 // device.
 type DeviceCharacteristic struct {
+	UUID
```

Comment:
I didn't catch this with the darwin support, but I wonder whether this would perhaps be better implemented as a getter?
The reason is that the SoftDevice has a special format for UUIDs that takes up just 3 or 4 bytes instead of the usual 16, and it would thus be possible to support them with less overhead (because many applications will just be looking for specific UUIDs and thus won't need to read the UUID field).

This could also be done in a separate PR, but seeing that the example heavily relies on UUIDs it may be best to do that in this PR.

(The same argument goes for `DeviceService.UUID`).

Reply:
I think that is a good idea, but perhaps best done in another PR.